x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_details/hooks/use_monitor_id.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useParams } from 'react-router-dom';
+import { ConfigKey } from '../../../../../../common/runtime_types';
+import { useSelectedMonitor } from './use_selected_monitor';
+
+export const useMonitorId = () => {
```

Comment:
Is there a scenario in which for a project monitor we might want to use the `monitor.id`?

If that's the case I think the name of this hook might be confusing down the line. I can see someone relying on this hook to get the _canonical_ ID, but getting the `CUSTOM_HEARTBEAT_ID` instead, generating confusion or bugs.

Maybe something like `useMonitorIdProject`, or `useMonitorInternalId`, or something related to the use of the ID itself?